R/sim_gs_n.R

```diff
@@ -505,6 +511,21 @@ create_cut <- function(...) {
   }
 }
 
+# Create cut functions from the design object
+cut_from_design <- function(x) {
+  if (is.null(x)) return()
+  k <- nrow(x$analysis)
+  res <- lapply(seq_len(k), function(i) {
+    if (is.null(x$input$analysis_time)) {
+      create_cut(target_event_overall = x$analysis$event[i])
+    } else {
+      create_cut(planned_calendar_time = x$analysis$time[i])
+    }
+  })
+  names(res) <- c(sprintf("ia%d", seq_len(k - 1)), "fa")
```

Comment:
Just FYI that technically these names are not required. The list of cutting functions is always accessed via numeric index inside of `sim_gs_n()`.